app.js

```diff
@@ -335,16 +335,28 @@ aws.initConfig(awsProps)
                     const id = req.params.id;
                     storageHandler.expandId(id)
                         .then(result => {
+                            const config = JSON.parse(result.config);
                             const metadata = {
-                                ogDescription: result.metadata ? result.metadata.description.S : null,
-                                ogAuthor: result.metadata ? result.metadata.author.S : null,
-                                ogTitle: result.metadata ? result.metadata.title.S : null
+                                ogDescription: result.special_metadata ? result.special_metadata.description.S : null,
+                                ogAuthor: result.special_metadata ? result.special_metadata.author.S : null,
+                                ogTitle: result.special_metadata ? result.special_metadata.title.S : "Compiler Explorer"
                             };
+                            if (!metadata.ogDescription) {
+                                if (result.metadata && result.metadata.single_code && result.metadata.single_code.M) {
+                                    const lang = languages[result.metadata.single_code.M.language.S];
+                                    metadata.ogDescription = result.metadata.single_code.M.source.S;
+                                    if (lang) {
+                                        metadata.ogTitle += ` - ${lang.name}`;
+                                    }
+                                }
+                            } else if (metadata.ogAuthor && metadata.ogAuthor !== '.') {
```

Comment:
What's the significance of the author of '.' ?

Reply:
I have not been able to create an empty string field for the db on the ce tool, so when no author is wanted, a . is set there. Please please feel free to fix it:
https://github.com/mattgodbolt/compiler-explorer-image/blob/master/bin/ce#L453

Reply:
Got it: just wondering! I'm cool with this! I just didn't know. Maybe a comment to explain, but really now I know I'm happy.